### PR TITLE
chore(weave): Ability to pivot on Calls Table

### PR DESCRIPTION
**Pull request by tssweeney**
Introduces the concept of a pivot table on the Calls Page. Currently we only show this option if all the items are from the `Evaluate-evalute` op, but that is just to keep the surface area small for now.

This is a bit exploratory and leads into the compare view.

Screenshot 2024-01-25 at 00 54 34

### CODE REVIEW

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsPage.tsx [hunk 1]
```diff
@@ -100,14 +103,13 @@ export const CallsTable: React.FC<{
...
   }, [effectiveFilter, orm]);
-  const runs = useRunsWithFeedback(
-    {
-      entityName: props.entity,
-      projectName: props.project,
-      streamName: 'stream',
-    },
-    lowLevelFilter
-  );
+  const streamId = {
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsPage.tsx:106]: Maybe just call this stream? I wouldn't expect something named Id to be an object.

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsPage.tsx [hunk 2]
```diff
@@ -152,6 +154,54 @@ export const CallsTable: React.FC<{
...
+            ...spec,
+          },
+        });
+      }
+    },
+    [filter, setFilter]
+  );
+
+  const qualifiesForPivoting = useMemo(() => {
+    const shownSpanNames = Array.from(
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsPage.tsx:192]: _.uniq might be more readable

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/PivotRunsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+          },
+        }}>
+        <Typography
+          style={{width: '38px', textAlign: 'center', flex: '0 0 auto'}}>
+          Pivot
+        </Typography>
+        <ListItem>
+          <FormControl fullWidth>
+            <Autocomplete
+              size={'small'}
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/PivotRunsTable.tsx:114]: I like to save characters and just use size="small"

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/PivotRunsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+        open={snackOpen}
+        autoHideDuration={2000}
+        onClose={() => {
+          setSnackOpen(false);
+        }}
+        message="Only 2 rows can be selected at a time."
+      />
+      <StyledDataGrid
+        columnHeaderHeight={40}
+        // apiRef={apiRef}
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/PivotRunsTable.tsx:294]: Remove commented out props?

**Review by jamie-rasmussen — APPROVED** [review]

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsPage.tsx (hunk 1 above)
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsPage.tsx:106]: 👍

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsPage.tsx (hunk 2 above)
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsPage.tsx:192]: 👍

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/PivotRunsTable.tsx (hunk 3 above)
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/PivotRunsTable.tsx:114]: 👍

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/PivotRunsTable.tsx (hunk 4 above)
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/PivotRunsTable.tsx:294]: 👍

**Pull request merged**